packages/amplication-server/src/core/assistant/assistantFunctions.service.ts

```diff
@@ -26,11 +26,90 @@ import { ResourceService } from "../resource/resource.service";
 import { AssistantContext } from "./dto/AssistantContext";
 import { EnumAssistantFunctions } from "./dto/EnumAssistantFunctions";
 import { MessageLoggerContext } from "./assistant.service";
+import { PermissionsService } from "../permissions/permissions.service";
+import { AuthorizableOriginParameter } from "../../enums/AuthorizableOriginParameter";
+import { get } from "lodash";
 
 export const MESSAGE_UPDATED_EVENT = "assistantMessageUpdated";
 
 export const PLUGIN_LATEST_VERSION_TAG = "latest";
 
+const FUNCTION_PERMISSIONS: {
```

Comment:
Since there is so much repetition here, this can be refactored to look nicer, and the top combinations can be reused (ResourceId+"serviced" and ProjectId+"projected"), etc.
Not a blocker to approve but a DRY thing.

Reply:
I decided to write this this way to force the developer of any future function to add the permissions settings
otherwise the type check will break

the type [key in EnumAssistantFunctions] requires that all enum members will be listed as properties of the object